libraries/c_sdk/standard/ble/src/iot_ble_gap.c

```diff
@@ -876,9 +900,63 @@ BTStatus_t IotBle_Init( void )
         }
     }
 
+    /*
+     * If a BLE device name is provided in the config, use it to set the local BLE device name.
+     */
+
+    if( ( pDeviceName != NULL ) && ( strcmp( pDeviceName, "" ) != 0 ) )
+    {
+        status = IotBle_SetDeviceName( pDeviceName, strlen( pDeviceName ) );
+    }
+
+    return status;
+}
+
+BTStatus_t IotBle_SetDeviceName( const char * pName,
+                                 size_t length )
+{
+    BTStatus_t status;
+
+    if( length <= IOT_BLE_MAX_DEVICE_NAME_LENGTH )
+    {
+        /* Copy the device name to a buffer, so that device name does not change between
+         * stack disable-enable cycles.
+         */
+        strncpy( bleDeviceName, pName, length );
+        bleDeviceName[ length + 1 ] = '\0';
```

Comment:
***This one is important:
```suggestion
        bleDeviceName[ length ] = '\0';
```
Given length is one-indexed, but the indexing is zero-indexed, this could go out of bounds.

Reply:
Buffer size is IOT_BLE_MAX_DEVICE_NAME_LENGTH + 1  (including null terminator) so it will not go out of bounds.

Reply:
Right, but say `length = IOT_BLE_MAX_DEVICE_NAME_LENGTH` and for sake of example max equals `4`. So then there's `IOT_BLE_MAX_DEVICE_NAME_LENGTH + 1 = 5` items alloced. This line could try to index 
`i = length + 1` --> `i = 4 + 1` -- > `i = 5`.  Then we're trying to index `array[5]` but that array only has 5 items i.e. max index of `array[4]`

Reply:
That was updated to ` bleDeviceName[ length ]` in one of the commits in PR. Could you check the latest version of PR.

Reply:
I see it! Awesome